vscode/src/state.ts

```diff
@@ -248,6 +248,93 @@ export class DeviceScriptExtensionState extends JDEventSource {
         )
     }
 
+    async showQuickPickBoard(title: string) {
+        const { boards } = this.devtools
+
+        // find device on the board
+        const devices = this.bus.devices({
+            serviceClass: SRV_DEVICE_SCRIPT_MANAGER,
+        })
+        const deviceItems = devices
+            .map(device => ({
+                device,
+                board: boards.find(
+                    b => parseAnyInt(b.productId) === device.productIdentifier
+                ),
+            }))
+            .filter(({ board }) => !!board)
+            .map(({ device, board }) => ({
+                data: { board },
+                description: board.devName,
+                label: device.shortId,
+                detail: "connected",
+            }))
+        const res = await vscode.window.showQuickPick(
+            <TaggedQuickPickItem<{ board?: DeviceConfig }>[]>[
+                ...deviceItems,
+                deviceItems && {
+                    kind: vscode.QuickPickItemKind.Separator,
+                },
+                {
+                    label: "Help me choose...",
+                    detail: "Identify your board or find where to get one.",
+                },
+                ...boards.map(board => ({
+                    data: { board },
+                    label: board.devName,
+                    description: board.id,
+                    detail: board.$description,
+                })),
+            ].filter(i => !!i),
+            {
+                title,
+                canPickMany: false,
+                matchOnDetail: true,
+                matchOnDescription: true,
+            }
+        )
+        if (!res) return // user escaped
+
+        if (!res.data?.board) {
+            openDocUri("devices")
+            return undefined
+        }
+
+        return res.data.board
+    }
+
+    async configureHardware(editor: vscode.TextEditor) {
+        await this.devtools.start()
+        if (!this.devtools.connected) return
+        const { boards } = this.devtools
+
+        const document = editor.document
+
+        // first identify the board
+        const { boardimport } =
+            /from "@dsboard\/(?<board>[^"]+)/.exec(document.getText())
+                ?.groups || {}
+        let board = boards.find(b => b.id === boardimport)
+        if (!board) {
+            await this.devtools.refreshSpecs()
+            board = await this.showQuickPickBoard(
+                "What kind of device are you programming?"
+            )
+            if (!board) return
+
+            // insert missing import
+            await editor.edit(editBuilder => {
+                editBuilder.insert(
+                    document.positionAt(0),
+                    `import { pins } from "@dsboard/${board.id}:\n`
```

Comment:
```suggestion
                    `import { pins, board } from "@dsboard/${board.id}"\n`
```

typo, but also I think we can import `board` constant - it has references to built-in services (though this doesn't work yet)